ch3: no `pthread_mutexattr_setpshared` on openbsd

more openbsd porting!   OpenBSD pthread does not have mutexattr_setpshared so we get the below warning at compile time and then a link failure

```
 CC       src/mpid/ch3/channels/nemesis/src/lib_libmpi_la-ch3_init.lo
../src/mpid/ch3/channels/nemesis/src/ch3_init.c:72:17: warning: implicit declaration of function 'pthread_mutexattr_setpshared' is invalid in C99 [-Wimplicit-function-declaration]
    mutex_err = pthread_mutexattr_setpshared(&attr, PTHREAD_PROCESS_SHARED);
                ^
1 warning generated.
   CC       src/mpid/ch3/channels/nemesis/src/lib_libmpi_la-ch3_win_fns.lo
../src/mpid/ch3/channels/nemesis/src/ch3_win_fns.c:573:13: warning: implicit declaration of function 'pthread_mutexattr_setpshared' is invalid in C99 [-Wimplicit-function-declaration]
            MPIDI_CH3I_SHM_MUTEX_INIT(*win_ptr);
            ^
../src/mpid/ch3/channels/nemesis/include/mpidi_ch3_impl.h:112:18: note: expanded from macro 'MPIDI_CH3I_SHM_MUTEX_INIT'
        pt_err = pthread_mutexattr_setpshared(&attr, PTHREAD_PROCESS_SHARED);           \
                 ^
1 warning generated.
```

--- a/src/mpid/ch3/channels/nemesis/include/mpidi_ch3_impl.h
+++ b/src/mpid/ch3/channels/nemesis/include/mpidi_ch3_impl.h
@@ -104,20 +104,8 @@ int MPIDI_CH3_SHM_Win_free(MPIR_Win **win_ptr);
 #define MPIDI_CH3I_SHM_MUTEX_INIT(win_ptr)                                              \
     do {                                                                                \
         int pt_err;                                                                     \
-        pthread_mutexattr_t attr;                                                       \
-                                                                                        \
-        pt_err = pthread_mutexattr_init(&attr);                                         \
-        MPIR_ERR_CHKANDJUMP1(pt_err, mpi_errno, MPI_ERR_OTHER, "**pthread_mutex",       \
-                             "**pthread_mutex %s", strerror(pt_err));                   \
-        pt_err = pthread_mutexattr_setpshared(&attr, PTHREAD_PROCESS_SHARED);           \
-        MPIR_ERR_CHKANDJUMP1(pt_err, mpi_errno, MPI_ERR_OTHER, "**pthread_mutex",       \
-                             "**pthread_mutex %s", strerror(pt_err));                   \
-        pt_err = pthread_mutex_init((win_ptr)->shm_mutex, &attr);                       \
-        MPIR_ERR_CHKANDJUMP1(pt_err, mpi_errno, MPI_ERR_OTHER, "**pthread_mutex",       \
-                             "**pthread_mutex %s", strerror(pt_err));                   \
-        pt_err = pthread_mutexattr_destroy(&attr);                                      \
-        MPIR_ERR_CHKANDJUMP1(pt_err, mpi_errno, MPI_ERR_OTHER, "**pthread_mutex",       \
-                             "**pthread_mutex %s", strerror(pt_err));                   \
+        MPL_proc_mutex_create((win_ptr)->shm_mutex, &pt_err);                           \
+        MPIR_ERR_CHECK(pt_err);                                                         \
     } while (0);
 
 #define MPIDI_CH3I_SHM_MUTEX_DESTROY(win_ptr)                                           \

--- a/src/mpid/ch3/channels/nemesis/src/ch3_init.c
+++ b/src/mpid/ch3/channels/nemesis/src/ch3_init.c
@@ -62,17 +62,7 @@ static int split_type(MPIR_Comm * user_comm_ptr, int stype, int key,
 
 int MPIDI_CH3I_Shm_supported(void)
 {
-    int mutex_err;
-    pthread_mutexattr_t attr;
-
-    /* Test for PTHREAD_PROCESS_SHARED support.  Some platforms do not support
-     * this capability even though it is a part of the pthreads core API (e.g.,
-     * FreeBSD does not support this as of version 9.1) */
-    pthread_mutexattr_init(&attr);
-    mutex_err = pthread_mutexattr_setpshared(&attr, PTHREAD_PROCESS_SHARED);
-    pthread_mutexattr_destroy(&attr);
-
-    return !mutex_err;
+    return MPL_proc_mutex_enabled();
 }
 
 static MPIR_Commops comm_fns = {

